docs/tutorial/t-spark-monitoring.md

```diff
@@ -0,0 +1,350 @@
+loki:grafana-dashboard              grafana:grafana-dashboard    grafana_dashboard      regular  
+loki:grafana-source                 grafana:grafana-source       grafana_datasource     regular  
+loki:metrics-endpoint               prometheus:metrics-endpoint  prometheus_scrape      regular  
+loki:replicas                       loki:replicas                loki_replica           peer     
+prometheus:grafana-dashboard        grafana:grafana-dashboard    grafana_dashboard      regular  
+prometheus:grafana-source           grafana:grafana-source       grafana_datasource     regular  
+prometheus:prometheus-peers         prometheus:prometheus-peers  prometheus_peers       peer     
+traefik:ingress                     alertmanager:ingress         ingress                regular  
+traefik:ingress                     catalogue:ingress            ingress                regular  
+traefik:ingress-per-unit            loki:ingress                 ingress_per_unit       regular  
+traefik:ingress-per-unit            prometheus:ingress           ingress_per_unit       regular  
+traefik:metrics-endpoint            prometheus:metrics-endpoint  prometheus_scrape      regular  
+traefik:peers                       traefik:peers                traefik_peers          peer     
+traefik:traefik-route               grafana:ingress              traefik_route          regular  
+```
+
+At this point, the observability stack has been deployed but Charmed Spark knows nothing about it. Since Charmed Spark comes with a built in spark metrics exporter that exports metric logs to a Prometheus gateway, we need to deploy a Prometheus gateway and then add Spark options to connect to it. The Prometheus gateway in turn will then be integrated with Prometheus. Let's deploy `prometheus-pushgateway-k8s` charm and integrate it with `prometheus` charm.
+
+```bash
+juju deploy prometheus-pushgateway-k8s --channel edge
+
+juju integrate prometheus-pushgateway-k8s prometheus
+```
+
+Now, for Spark to be able to access the Prometheus gateway, we need the gateway address and port. Let's export them as environment variables so that they can be used later.
+
+```shell
+# Get prometheus gateway IP
+export PROMETHEUS_GATEWAY=$(juju status --format=json | jq -r '.applications."prometheus-pushgateway-k8s".address') 
+
+export PROMETHEUS_PORT=9091
+```
+
+Now that we have Prometheus gateway IP adress and port, let's create a new service account in `cos` namespace with all configuration options as `spark` service account in `spark` namespace with a few additional configuration options related to Prometheus gateway.
+
+```shell
+# Get config from old service account and store in a file
+spark-client.service-account-registry get-config \
+  --username spark --namespace spark > properties.conf
+
+# Create a new service account and load configurations from the file
+spark-client.service-account-registry create \
+  --username spark --namespace cos \
+  --properties-file properties.conf
```

Comment:
Also in this case, it should not be need to deploy spark-jobs and service account to the cos namespace. Actually, maybe it is better to have a namespace for observability stuff, and another way for the user to run spark jobs

Reply:
During my testing, using different namespaces resulted in the DNS of the pods (running in Juju model's namespace) not being resolved from the pods running in spark serviceaccount's namespace. That's the reason I kept dedicated namespaces for each section of the tutorial.